src/wayland/selection/seat_data.rs

```diff
@@ -4,14 +4,14 @@ use wayland_server::{Client, DisplayHandle, Resource};
 
 use crate::utils::IsAlive;
 
-use super::device::SelectionDevice;
+use super::device::{DataControlDevice, SelectionDevice};
 use super::offer::{OfferReplySource, SelectionOffer};
 use super::{SelectionHandler, SelectionTarget};
 
 /// Seat data used to handle regular selection operations.
 ///
-/// The data is shared accross primary, data device, and data control selections.
-pub struct SeatData<U: Clone + Sync + Send + 'static> {
+/// The data is shared across primary, data device, and data control selections.
+pub(crate) struct SeatData<U: Clone + Sync + Send + 'static> {
```

Comment:
I think it's a part of the public API? At least all the `Data` was usually a public API.

Reply:
The `seat_data` module is private, and the type is not reexported. So I just marked it as pub(crate) to make that fact more clear.